context/app/static/js/pages/Dataset/Dataset.jsx

```diff
@@ -134,7 +138,11 @@ function DatasetDetail({ assayMetadata, vitData, hasNotebook, visLiftedUUID }) {
     visualization: Boolean(vitData),
     protocols: Boolean(protocol_url),
     metadata: Boolean(Object.keys(combinedMetadata).length),
-    files: true,
+    files: !isAuthenticated && accessType === 'Public',
```

Comment:
I would show files if `metadata.files` exists or has a length. I forget if it's an array.